File: 05-Lista_simplemente_ligada/cpp/main.cpp
#include "simpleLinkedList.h"

using namespace std;

int main() {
    SimpleLinkedList<int> *listaLigada = new SimpleLinkedList<int>();

    cout << boolalpha;

    cout << *listaLigada << endl;

    // Esta vacia
    cout << "\nCompureba que la lista este vacia" << endl;
    cout << listaLigada->isEmpty() << endl;

    // Tamaño
    cout << "\nTamanio" << endl;
    cout << listaLigada->getSize() << endl;

    // Agregar al inicio
    cout << "\nAgrega 12 al inicio" << endl;
    listaLigada->addFirst(12);
    cout << *listaLigada << endl;
    cout << "Agrega 32 al inicio" << endl;
    listaLigada->addFirst(32);
    cout << *listaLigada << endl;
    cout << "Agrega 61 al inicio" << endl;
    listaLigada->addFirst(61);
    cout << *listaLigada << endl;

    // Agregar al final
    cout << "\nAgrega 28 al final" << endl;
    listaLigada->addLast(28);
    cout << *listaLigada << endl;

    // Agregar despues de
    cout << "\nAgregar 55 despues de 12" << endl;
    listaLigada->addAfter(12, 55);
    cout << *listaLigada << endl;

    // Elimina en la posicion
    cout << "\nElimina en la posicion 2" << endl;
    listaLigada->deleteAt(2);
    cout << *listaLigada << endl;

    // Elimina al inicio
    cout << "\nElimina al inicio" << endl;
    listaLigada->deleteStart();
    cout << *listaLigada << endl;

    // Elimina al final
    cout << "\nElimina al final" << endl;
    listaLigada->deleteLast();
    cout << *listaLigada << endl;

    // Busca el valor
    cout << "\nBusca el 55" << endl;
    cout << "Resultado: " << listaLigada->search(55) << endl;

    // Actualiza el 61 a 10
    cout << "\nActualiza el 32 a 10" << endl;
    listaLigada->update(32, 10);
    cout << *listaLigada << endl;

    delete listaLigada;
}

File: 05-Lista_simplemente_ligada/python/main.py
from simpleLinkedList import SimpleLinkedList


listaLigada = SimpleLinkedList();
print(listaLigada);

# Esta vacia
print("\nCompureba que la lista este vacia");
print(listaLigada.isEmpty());

# Tamaño
print("\nTamaño");
print(listaLigada.getSize());

# Agregar al inicio
print("\nAgrega 12 al inicio");
listaLigada.addFirst(12);
print(listaLigada);
print("Agrega 32 al inicio");
listaLigada.addFirst(32);
print(listaLigada);
print("Agrega 61 al inicio");
listaLigada.addFirst(61);
print(listaLigada);

# Agregar al final
print("\nAgrega 28 al final");
listaLigada.addLast(28);
print(listaLigada);

# Agregar despues de
print("\nAgregar 55 despues de 12");
listaLigada.addAfter(12, 55);
print(listaLigada);

# Elimina en la posicion
print("\nElimina en la posicion 2");
listaLigada.deleteAt(2);
print(listaLigada);

# Elimina al inicio
print("\nElimina al inicio");
listaLigada.deleteStart();
print(listaLigada);

# Elimina al final
print("\nElimina al final");
listaLigada.deleteLast();
print(listaLigada);

# Busca el valor
print("\nBusca el 55");
print("Resultado: " + str(listaLigada.search(55)));

# Actualiza el 61 a 10
print("\nActualiza el 32 a 10");
listaLigada.update(32, 10);
print(listaLigada);
